src/WikiService.Business/Commands/Rubric/Edit/EditRubricHandler.cs

```diff
@@ -26,6 +26,107 @@ public class EditRubricHandler : IRequestHandler<EditSpecificRubricRequest, bool
     private readonly IHttpContextAccessor _httpContextAccessor;
     private readonly IMemoryCache _cache;
 
+    private Task<ValueTuple<int, Guid>[]> GetPositionsAsync(Guid? parentId)
+    {
+      return _provider.Rubrics
+        .Where(x => x.ParentId == parentId)
+        .Select(x => new ValueTuple<int, Guid>(x.Position, x.Id))
+        .ToArrayAsync();
+    }
+
+    private async Task<bool> EditPositionsAsync(
+      ValueTuple<int, Guid>[] positions,
+      Guid? parentId)
+    {
+      List<DbRubric> rubrics = await _provider.Rubrics.Where(x => x.ParentId == parentId).ToListAsync();
+
+      foreach (DbRubric rubric in rubrics)
+      {
+        int position = positions.FirstOrDefault(x => x.Item2 == rubric.Id).Item1;
+        rubric.Position = position == 0
+          ? rubric.Position
+          : position;
+      }
+
+      return true;
+    }
+
+    private async Task<bool> ChangePositionAsync(Guid id, int position, Guid? parentId)
+    {
+      ValueTuple<int, Guid>[] positions = await GetPositionsAsync(parentId);
+
+      Array.Sort(positions);
+
+      for (int i = 0; i < positions.Length; i++)
+      {
+        if (positions[i].Item2 == id)
+        {
+          if (positions[i].Item1 < position)
+          {
+            int j = i + 1;
+            while (j < positions.Length && positions[j].Item1 <= position)
+            {
+              positions[j].Item1--;
+              j++;
+            }
+          }
+          else
+          {
+            int j = i - 1;
+            while (j >= 0 && positions[j].Item1 >= position)
+            {
+              positions[j].Item1++;
+              j--;
+            }
+          }
+
+          positions[i].Item1 = position;
+          break;
+        }
+      }
```

Comment:
maybe move this algorithm to separate class? helper or smth like this

Reply:
would it be OK for feature-folder?

Reply:
I'm not sure, since this algorithm works with the database, which means that then only part of the algorithm can be transferred, which can be confusing in the future.